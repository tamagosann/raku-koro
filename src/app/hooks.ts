import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './store';

// useDispatchとかuseSelectorの代わりにこっちを使う
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
