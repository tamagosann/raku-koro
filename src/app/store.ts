import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import dailyInfectionRuducer from "../features/graphs/dailyInfectionSlice";
import prefectureRuducer from "../features/common/prefectureSlice";
import dailyDeadRuducer from "../features/graphs/dailyDeadSlice";
import totalCoronaRuducer from "../features/graphs/totalCoronaSlice";
import totalDethRuducer from "../features/graphs/totalDethSlice";
import vaccinationRuducer from "../features/graphs/vaccinationSlice";
import bedOccupancyRateRuducer from "../features/graphs/bedOccupancyRateSlice";
import dailyPositiveRuducer from "../features/graphs/dailyPositiveSlice";

export const store = configureStore({
  reducer: {
    dailyInfection: dailyInfectionRuducer,
    prefecture: prefectureRuducer,
    dailyDead: dailyDeadRuducer,
    totalCorona: totalCoronaRuducer,
    totalDeth: totalDethRuducer,
    vaccination: vaccinationRuducer,
    bedOccupancyRate: bedOccupancyRateRuducer,
    dailyPositive: dailyPositiveRuducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
