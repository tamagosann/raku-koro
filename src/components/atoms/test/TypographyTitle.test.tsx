import React from 'react';
import {
  render,
  cleanup,
  screen,
  fireEvent,
  waitFor,
  within,
  getByRole,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { afterEach, expect } from '@jest/globals';
import { store } from '../../../app/store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { TypographyTitle } from '../index';
import { PropTypes } from '@material-ui/core';
import { Variant } from '@material-ui/core/styles/createTypography';

describe('TypographyTitleコンポーネント', () => {
  // デモProps
  interface TypographyPropsInterface {
    children: string;
    variant: Variant;
    align: PropTypes.Alignment;
    className?: string;
  }

  const TypographyProps: TypographyPropsInterface = {
    children: 'テスト',
    variant: 'h1',
    align: 'center',
    className: 'testClass',
  };

  // 各テストの直前に実行される（コンポーネントの取得）
  beforeEach(() => {
    render(
      <Provider store={store}>
        <TypographyTitle
          children={TypographyProps.children}
          variant={TypographyProps.variant}
          align={TypographyProps.align}
          className={TypographyProps.className}
        />
      </Provider>
    );
  });

  // 各処理が通った後にアンマウントさせる
  afterEach(() => cleanup());

  describe('jest renderer test', () => {
    test('render correctly', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <TypographyTitle
              children={TypographyProps.children}
              variant={TypographyProps.variant}
              align={TypographyProps.align}
              className={TypographyProps.className}
            />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('h1タグのchildren私たPropsの確認', () => {
    test('h1タグのchildren私たPropsの確認テスト', () => {
      const headingTitle = screen.getByRole('heading');
      expect(headingTitle.textContent).toBe('テスト');
    });
  });
});
