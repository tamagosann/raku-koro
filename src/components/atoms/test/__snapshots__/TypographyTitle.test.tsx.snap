// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographyTitleコンポーネント jest renderer test render correctly 1`] = `
<h1
  className="MuiTypography-root testClass MuiTypography-h1 MuiTypography-alignCenter"
  style={
    Object {
      "marginBottom": "40px",
    }
  }
>
  テスト
</h1>
`;
